integration/all/trace_network_test.go

```diff
@@ -105,20 +103,38 @@ func TestTraceNetwork(t *testing.T) {
 
 			// TODO: Handle once we can get ContainerImageName from docker
 			if isDockerRuntime {
-				expectedEntries[1].Event.Runtime.ContainerImageName = ""
+				expectedEntries[1].Runtime.ContainerImageName = ""
 			}
 
-			normalize := func(e *networkTypes.Event) {
-				// Docker and CRI-O use a custom container name composed, among
-				// other things, by the pod UID. We don't know the pod UID in
-				// advance, so we can't match the exact expected container name.
-				if *containerRuntime == ContainerRuntimeDocker || *containerRuntime == ContainerRuntimeCRIO {
-					cn := e.K8s.ContainerName
-					if strings.HasPrefix(e.Runtime.ContainerName, fmt.Sprintf("k8s_%s_%s_%s_", cn, cn, ns)) {
-						e.Runtime.ContainerName = cn
-					}
-				}
+			if DefaultTestComponent == IgTestComponent {
+				expectedEntries[0].Event = BuildBaseEvent(ns,
+					WithRuntimeMetadata(containerRuntime),
+					WithContainerImageName("docker.io/library/busybox:latest", isDockerRuntime),
+					WithPodLabels("test-pod", ns, isCrioRuntime),
+				)
+
+				expectedEntries[1].K8s.PodLabels = addPodLabels()
+				expectedEntries[1].Runtime.ContainerName = "nginx-pod"
+				expectedEntries[1].Runtime.RuntimeName = eventtypes.String2RuntimeName(containerRuntime)
+			} else if DefaultTestComponent == InspektorGadgetTestComponent {
+				expectedEntries[0].Event = BuildBaseEventK8s(ns, WithContainerImageName("docker.io/library/busybox:latest", isDockerRuntime))
+				expectedEntries[0].PodIP = testPodIP
+				expectedEntries[0].PodLabels = map[string]string{"run": "test-pod"}
```

Comment:
```suggestion
				expectedEntries[0].Event = BuildBaseEventK8s(ns,
					WithContainerImageName("docker.io/library/busybox:latest", isDockerRuntime),
					WithPodLabels("test-pod", ns, isCrioRuntime),
				)
				expectedEntries[0].PodIP = testPodIP
```

Reply:
`WithPodLabels` sets `expectedEntries[0].K8s.PodLabels`, whereas we need to assign to `expectedEntries[0].PodLabels`